### ISSUE TITLE
Distilling the Knowledge in a Neural Network

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#6]
# Distilling the Knowledge in a Neural Network[[link](https://arxiv.org/abs/1503.02531)]

### Distilling : ensemble의 정보를 single model로 이전하는 과정

+ ensemble을 하는 이유 : NN은 성능이 좋지만 파라미터가 많아서 오버피팅이 되기 쉬움
  + 여러개의 single NN을 ensemble하여 일반화 시키면 성능이 좋아짐
  + 여기서 여러개의 single model의 architecture를 다르게 하는것 보다 init을 다르게 하는게 좋다고 함

+ ensemble의 단점 : 계산(추론)시간이 오래 걸리고 저장공간을 많이 필요로 한다
  + 만약 core갯수보다 model이 많으면 single model을 학습시키는 것 보다 더 느려짐
  + mobile같은 경우는 저장공간이 한정되어 있기 때문에 적용이 더 어려움
 
+ 즉, single shallow model을 만드는것이 목표

### 1. oversampling without label
+ label이 붙어있지 않은 데이터를 기존 ensemble model로 추론후 결과값을 label로 학습
+ ensemble model의 정보는 label로 이전됨

### 2. learning class logit
+ 1.번 방법처럼 데이터가 많은것도 중요하지만, 각 class의 확률 분포를 학습하면 더 좋을 것이다
+ single model을 학슬할 때 기존 데이터의 label이 아닌 ensemble model로 추론한 logit값을 학습함
+ 최근 논문에는 logit에 epsilon(noise)를 더하여 학습하면 더 좋다는 연구도 있음(정규화 효과)

### 3. using softmax(이 논문에서 다루는 방법) 
+ sigmoid는 각 class logit값의 합이 1이 아니기 때문에 ensemble model의 softmax로 나온 확률값을 학습
+ 확률값을 이용하여 학습하는것은 정규화의 역할도 한다.

## Temperature
softmax를 사용하면 
![image](https://user-images.githubusercontent.com/31437443/106423310-88a69f00-64a3-11eb-99f0-afd402076456.png)
+ T = 1 : 기존의 softmax 
  + ensemble model의 성능이 너무 좋으면 기존 label과 동일(=1)하게 되므로 Distilling의 의미가 없어짐
+ T가 높을 때 : smoothing이 심하게 되어 각 class 확률값이 비슷해지므로 확률 분포로써 의미가 없어짐
+ 적당한 T : 학습이 잘 이루어짐
+ 즉, 적당한 T를 찾는 것이 중요한데 논문에서 T를 찾는 방법이 나와있지 않아 grid하게 찾아본것이라고 생각함

## cost function 
기존의 cross entropy에서 indicator를 softmax값으로 변경한 Cross Entropy Cost Function

## conclusion
+ ensemble을 shallow single model로 이전하는것이 가능하다
  + hinton은 이를 진화하는 과정이라고 함
